hr_payroll_expense/models/hr_payslip.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# © 2016 Coninckx David (Open Net Sarl)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields, api
+
+
+class HrPayslip(models.Model):
+    _inherit = 'hr.payslip'
+
+    expenses = fields.One2many('hr.expense', 'slip_id',
+                               string='Expenses')
+
+    @api.multi
+    def compute_sheet(self):
+        # Detach expenses from the pay slips
+        ExpenseObj = self.env['hr.expense']
+        expenses = ExpenseObj.search([('slip_id', 'in', self.ids)])
+        if expenses:
+            expenses.write({'slip_id': False})
+
+        res = super(HrPayslip, self).compute_sheet()

```

Comment:
The super must be called before the slip id is set on the expenses (hr_payroll_expense) or on the invoices (hr_payroll_commission). It also must be called after the expenses or the invoices are detached from the employee's payslip. I know it's a be difficult to understand but it's a way to handle the case where 2 draft employee's payslip compute.